optimizely/config_manager.py

```diff
@@ -56,29 +56,67 @@ class StaticConfigManager(BaseConfigManager):
     def __init__(self,
                  datafile=None,
                  logger=None,
-                 error_handler=None):
+                 error_handler=None,
+                 skip_json_validation=False):
         """ Initialize config manager. Datafile has to be provided to use.
 
         Args:
             datafile: JSON string representing the Optimizely project.
             logger: Provides a logger instance.
             error_handler: Provides a handle_error method to handle exceptions.
+            skip_json_validation: Optional boolean param which allows skipping JSON schema
+                                  validation upon object invocation. By default
+                                  JSON schema validation will be performed.
         """
         super(StaticConfigManager, self).__init__(logger=logger, error_handler=error_handler)
         self._config = None
-        if datafile:
-          self._config = project_config.ProjectConfig(datafile, self.logger, self.error_handler)
+        self.validate_schema = not skip_json_validation
+        self.set_config(datafile)
+
+    def set_config(self, datafile):
+        """ Looks up and sets datafile and config based on response body.
+
+         Args:
+           datafile: JSON string representing the Optimizely project.
+         """
+
+        if self.validate_schema:
+            if not validator.is_datafile_valid(datafile):
+                self.logger.error(enums.Errors.INVALID_INPUT.format('datafile'))
+                return
+
+        error_msg = None
+        error_to_handle = None
+        config = None
+
+        try:
+            config = project_config.ProjectConfig(datafile, self.logger, self.error_handler)
+        except optimizely_exceptions.UnsupportedDatafileVersionException as error:
+            error_msg = error.args[0]
+            error_to_handle = error
+        except:
+            error_msg = enums.Errors.INVALID_INPUT.format('datafile')
+            error_to_handle = optimizely_exceptions.InvalidInputException(error_msg)
+        finally:
+            if error_msg:
+                self.logger.error(error_msg)
+                self.error_handler.handle_error(error_to_handle)
+                return
+
+        # TODO(ali): Add notification listener.
+        self._config = config
```

Comment:
We need to check whether the revision is different before setting it.

Reply:
We don't do so in other SDKs and I think we should respect the last modified timestamp (which is happening already).

Reply:
Ah you're right. The timestamp already takes care of that.

Reply:
Isn't check on entire content a better check? Also, you are not checking if new version is higher. I feel in JS SDK you have already gone through the entire effort and then throwing away the config. There is a new datafile available and the config manager should just respond to that by updating.
What do you think?

Reply:
Updating this.